tests/test_pdf_gaussexptail.py

```diff
@@ -0,0 +1,184 @@
+from zfit.models.physics import GaussExpTail, GeneralizedGaussExpTail
+
+mu = -0.3
+sigma = 1.1
+
+sigmal = 1.1
+sigmar = 1.1
+
+alphal = 0.8
+nl = alphal
+
+alphar = 1.4
+nr = alphar
+
+bounds = (-6, 4)
+lbounds = (bounds[0], mu)
+rbounds = (mu, bounds[1])
+
+def _numpy_gaussexptail_pdf(x, mu, sigma, alpha, n):
+    t = (x - mu) / sigma * np.sign(alpha)
+    abs_alpha = np.abs(alpha)
+    if t > -abs_alpha:
+        return np.exp(-0.5 * t ** 2)
+    else:
+        return np.exp(-0.5 * abs_alpha ** 2) * np.exp(n * (t + abs_alpha))
+
+numpy_gaussexptail_pdf = np.vectorize(_numpy_gaussexptail_pdf, excluded=["mu", "sigma", "alpha", "n"])
+
+def _numpy_generalizedgaussexptail_pdf(x, mu, sigmal, alphal, nl, sigmar, alphar, nr):
+    if x < mu:
+        return _numpy_gaussexptail_pdf(x, mu, sigmal, alphal, nl)
+    else:
+        return _numpy_gaussexptail_pdf(x, mu, sigmar, -alphar, nr)
+
+numpy_generalizedgaussexptail_pdf = np.vectorize(_numpy_generalizedgaussexptail_pdf, excluded=["mu", "sigmal", "alphal", "nl", "sigmar", "alphar", "nr"])
+
+
+def _gaussexptail_params_factory(name_add=""):
+    mu_ = zfit.Parameter(f"mu_gaussexptail{name_add}", mu)
+    sigma_ = zfit.Parameter(f"sigma_gaussexptail{name_add}", sigma)
+    alphal_ = zfit.Parameter(f"alphal_gaussexptail{name_add}", alphal)
+    return {"mu": mu_, "sigma": sigma_, "alpha": alphal_}
+
+tester.register_pdf(pdf_class=GaussExpTail, params_factories=_gaussexptail_params_factory)
+
+
+
+def sample_testing(pdf):
+    sample = pdf.sample(n=1000, limits=(-0.5, 1.5))
+    sample_np = sample.numpy()
+    assert not any(np.isnan(sample_np))
+
+
+def eval_testing(pdf, x):
+    probs = pdf.pdf(x, norm=False)
+    assert probs.shape.rank == 1
+    assert probs.shape[0] == x.shape[0]
+    probs = probs.numpy()
+    assert not np.any(np.isnan(probs))
+    return probs
+
+
+def test_gaussexptail_integral():
+    obs = zfit.Space("x", limits=bounds)
+
+    mu_ = zfit.Parameter("mu", mu)
+    sigma_ = zfit.Parameter("sigma", sigma)
+    alphal_ = zfit.Parameter("alphal", alphal)
+
+    gaussexptail = GaussExpTail(obs=obs, mu=mu_, sigma=sigma_, alpha=alphal_)
+    int_limits = (-1, 3)
+    integral_numeric = gaussexptail.numeric_integrate(limits=int_limits, norm=False).numpy()
+    integral = gaussexptail.analytic_integrate(limits=int_limits, norm=False).numpy()
+    integral_scipy = integrate.quad(numpy_gaussexptail_pdf, *int_limits, args=(mu, sigma, alphal, nl))[0]
+
+    assert pytest.approx(integral_numeric, 1e-5) == integral
+    assert pytest.approx(integral_scipy, 1e-5) == integral
+
+    rnd_limits = sorted(list(np.random.uniform(*bounds, 13)) + list(bounds))
+    integrals = [
+        gaussexptail.integrate((low, up), norm=False)
+        for low, up in zip(rnd_limits[:-1], rnd_limits[1:])
+    ]
+
+    integral = np.sum(integrals)
+    integral_full = gaussexptail.integrate(bounds, norm=False).numpy()
+    assert pytest.approx(integral_full.item()) == integral.item()
```

Comment:
what is this `.item()` exactly and why is it needed? It should work without, right?

Reply:
`.item()` in this case gets the element from an array with just 1 element. https://numpy.org/doc/stable/reference/generated/numpy.ndarray.item.html. I think without it, `pytest.approx` doesn't like the shapes.

Reply:
Ok, it's fine in this case. I'm removing it. I've had cases when we're comparing a zfit result with a numba_stats result using `pytest.approx` where this is required.